src/dev/flang/ast/Call.java

```diff
@@ -2479,7 +2479,7 @@ Expr resolveSyntacticSugar(Resolution res, AbstractFeature outer)
   {
     Expr result = this;
     //    if (true) return result;
-    if (!Errors.any())
+    if (!Errors.any() && !isInheritanceCall())
```

Comment:
Please add a comment why this is not done for an inheritance call. 
I guess that for inherting from a boolean operator (`f : true.prefix !`) , you will get an error anyways since these are not constructors (better check:-), while for a numeric literal (`f : i32 42`) we do not want this replaced by a `NumLiteral`.